test/es6/toStringTag.js

```diff
@@ -5,299 +5,221 @@
+    { tag: "Error",             instance: new Error(),                           prototype: Error.prototype },
+    { tag: "Boolean",           instance: true,                                  prototype: Boolean.prototype },
+    { tag: "Number",            instance: 1,                                     prototype: Number.prototype },
+    { tag: "Date",              instance: new Date(),                            prototype: Date.prototype },
+    { tag: "RegExp",            instance: /a/,                                   prototype: RegExp.prototype },
+    { tag: "Object",            instance: {},                                    prototype: Object.prototype },
+];
 
-            assert.areEqual("WeakMap", (new WeakMap())[Symbol.toStringTag], "check String is WeakMap");
-            var o  = Object.getOwnPropertyDescriptor(WeakMap.prototype, Symbol.toStringTag);
-            assert.isFalse(o.writable,     "WeakMap @@toStringTag is not writable");
-            assert.isFalse(o.enumerable,   "WeakMap @@toStringTag is not enumerable");
-            assert.isTrue(o.configurable,  "WeakMap @@toStringTag is configurable");
+var argumentsInternalSlotBuiltIn = 
+    { tag: "Arguments",         instance: (function () { return arguments; })(), prototype: (function () { return arguments; })() }
 
-            assert.areEqual("Set", (new Set())[Symbol.toStringTag], "check String is Set");
-            var o  = Object.getOwnPropertyDescriptor(Set.prototype, Symbol.toStringTag);
-            assert.isFalse(o.writable,     "Set @@toStringTag is not writable");
-            assert.isFalse(o.enumerable,   "Set @@toStringTag is not enumerable");
-            assert.isTrue(o.configurable,  "Set @@toStringTag is configurable");
+var tests = [
+   {
+       name: "Built-in prototype @@toStringTag properties",
+       body: function () {
+            assert.areEqual("Symbol", Symbol.prototype[Symbol.toStringTag], "check String is Symbol");
+            var o  = Object.getOwnPropertyDescriptor(Symbol.prototype, Symbol.toStringTag);
+            assert.isFalse(o.writable,     "Symbol @@toStringTag is not writable");
+            assert.isFalse(o.enumerable,   "Symbol @@toStringTag is not enumerable");
+            assert.isTrue(o.configurable,  "Symbol @@toStringTag is configurable");
 
-            assert.areEqual("WeakSet", (new WeakSet())[Symbol.toStringTag], "check String is WeakSet");
-            var o  = Object.getOwnPropertyDescriptor(WeakSet.prototype, Symbol.toStringTag);
-            assert.isFalse(o.writable,     "WeakSet @@toStringTag is not writable");
-            assert.isFalse(o.enumerable,   "WeakSet @@toStringTag is not enumerable");
-            assert.isTrue(o.configurable,  "WeakSet @@toStringTag is configurable");
+            function checkBuiltInPrototype(builtInName, prototype) {
+                WScript.Echo("Testing " + builtInName);

```

Comment:
This is left over from debugging. I found it helped to diagnose some issues where the assert wouldn't fire and show the message, especially when we are testing multiple types of built ins. I can either leave it in and make it depend on WScript.Arguments[0] != "summary" or just take it out completely.